R/multilevelannotationstep2.R

```diff
@@ -21,21 +21,21 @@ compute_something <- function(m, curmchemdata, mass_defect_window, isop_res_md)
 #' @import plyr
 #' @export
 compute_chemscore <- function(j,
-                              chemids,
-                              mchemdata,
-                              isop_res_md,
-                              mass_defect_window,
-                              corthresh,
-                              global_cor,
-                              mzid,
-                              max_diff_rt,
-                              adduct_table,
-                              adduct_weights,
-                              filter.by,
-                              max_isp,
-                              MplusH.abundance.ratio.check,
-                              mass_defect_mode,
-                              outloc
+                              chemids = NA,
+                              mchemdata = NA,
+                              isop_res_md = NA,
+                              mass_defect_window = NA,
+                              corthresh = NA,
+                              global_cor = NA,
+                              mzid = NA,
+                              max_diff_rt = NA,
+                              adduct_table = NA,
+                              adduct_weights = NA,
+                              filter.by = NA,
+                              max_isp = NA,
+                              MplusH.abundance.ratio.check = NA,
+                              mass_defect_mode = NA,
+                              outloc = NA

```

Comment:
What is the point of passing `NA` by default? Given the quality of the original code, I'd say it may compute scores incorrectly instead of notifying that some parameters are missing if you forget to pass them.

Reply:
I had this added to make the parameters named, to make sure that they are being passed in the correct fields, because I don't want to rely on the ordering - with that number of parameters it get's easy to mix that up.

There is a way how to make sure that actually all named parameters are passed, but it makes the syntax a bit uglier.

I'm not sure how exactly named and unnamed function arguments in R work - but that is a good point to keep in mind!

Reply:
Edit: Seems like you can supply **by position or by name** according to here - than I should remove the default values indeed to make sure that everything is passed correctly!